runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/widgets/CredentialWidget/CustomCredentialSelectDialog.tsx

```diff
@@ -0,0 +1,94 @@
+import { FunctionComponent, useEffect, useState } from 'react';
+
+
+import { Box, Button, Dialog, DialogActions, DialogTitle, Typography } from '@mui/material';
+import { useRouter } from 'next/router';
+import { useSnackbar } from 'notistack';
+import { CredentialDto } from 'runbotics-common';
+
+import CredentialSelectSet from '#src-app/components/CredentialSelectSet';
+import If from '#src-app/components/utils/If';
+import { translate } from '#src-app/hooks/useTranslations';
+import { useDispatch, useSelector } from '#src-app/store';
+import { credentialsSelector, credentialsActions } from '#src-app/store/slices/Credentials/Credentials.slice';
+import { AddDialogContent, StyledButton } from '#src-app/views/process/ProcessConfigureView/ProcessCredentials/ProcessCredentials.styles';
+
+interface CustomCredentialSelectDialogProps {
+    isOpen: boolean;
+    handleClose: () => void;
+    templateName: string;
+    credential: CredentialDto | null;
+    setCredential: (credential: CredentialDto | null) => void;
+}
+
+export const CustomCredentialSelectDialog: FunctionComponent<CustomCredentialSelectDialogProps> = ({
+    isOpen, handleClose, templateName, credential, setCredential
+}) => {
+    const { enqueueSnackbar } = useSnackbar();
+    const { id: processId } = useRouter().query;
+    const dispatch = useDispatch();
+    const { allAvailable } = useSelector(credentialsSelector);
+    const [pickedCredential, setPickedCredential] = useState(null);
+    const handleDialogClose = () => {
+        handleClose();
+    };
+
+    const handleCredentialChange = (selectedCredential) => {
+        setPickedCredential(selectedCredential);
+    };
+
+    const handleSelectCredential = () => {
+        setCredential(pickedCredential);
+        handleClose();
+    };
+
+    useEffect(() => {
+        if (isOpen) {
+            console.log('fetching availablecredentials');
+            dispatch(credentialsActions.fetchAllCredentialsAssignedToProcess({ resourceId: String(processId) }));
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why exactly do we need to disable eslint rule for these deps?

Reply:
I think it is because it always shout that the dispatch should be included in dependencies

Reply:
exactly